src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -483,8 +483,15 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
 
     fun addFun(funSpec: FunSpec) = apply {
       if (kind == Kind.INTERFACE) {
-        requireExactlyOneOf(funSpec.modifiers, KModifier.ABSTRACT)
-        requireExactlyOneOf(funSpec.modifiers, KModifier.PUBLIC, KModifier.PRIVATE)
+
+        if(funSpec.body.isEmpty()) {
+          require(!funSpec.modifiers.contains(KModifier.PRIVATE), {"Abstract $kind function can not be private"})
+          requireNotNull(funSpec.returnType,
```

Comment:
It's actually fine to suppress the type, it will always default to `Unit`. When I add `Unit` explicitly, IDE warns me that it's redundant.

Also here, same thing about placing lambda outside the function parentheses.